External/FEXCore/Source/Interface/IR/Passes/DeadContextStoreElimination.cpp

```diff
@@ -608,7 +619,14 @@ bool RCLSE::Run(FEXCore::IR::IREmitter *IREmit) {
   // XXX: We don't do cross-block optimizations yet
   //CalculateControlFlowInfo(IREmit);
   bool Changed = false;
-  Changed |= RedundantStoreLoadElimination(IREmit);
+  
+  VerboseLog("Start!");
+  while(RedundantStoreLoadElimination(IREmit)) {
+    Changed = true;
+    DCE->Run(IREmit);
+    VerboseLog("%p: Changed, Re-running\n", this);
```

Comment:
If I'm right, this should never run more than twice. 

First pass will remove all the LoadContexts it can. Second pass will pickup and remove all the redundant StoreContexts

Reply:
In some cases, it runs 4-5 times, removing one or two more ops per iteration.

Reply:
Weird. Maybe I don't fully understand the flow of this pass.